src/contentsharecontroller/DefaultContentShareController.ts

```diff
@@ -52,15 +63,55 @@ export default class DefaultContentShareController implements ContentShareContro
   }
 
   pauseContentShare(): void {
-    this.mediaStreamBroker.toggleMediaStream(false);
+    if (this.mediaStreamBroker.toggleMediaStream(false)) {
+      this.forEachContentShareObserver(observer => {
+        Maybe.of(observer.contentShareDidPause).map(f => f.bind(observer)());
+      });
+    }
   }
 
   unpauseContentShare(): void {
-    this.mediaStreamBroker.toggleMediaStream(true);
+    if (this.mediaStreamBroker.toggleMediaStream(true)) {
+      this.forEachContentShareObserver(observer => {
+        Maybe.of(observer.contentShareDidUnpause).map(f => f.bind(observer)());
+      });
+    }
   }
 
   stopContentShare(): void {
     this.audioVideo.stop();
     this.mediaStreamBroker.cleanup();
   }
+
+  addContentShareObserver(observer: ContentShareObserver): void {
+    this.observerQueue.add(observer);
+  }
+
+  removeContentShareObserver(observer: ContentShareObserver): void {
+    this.observerQueue.delete(observer);
+  }
+
+  forEachContentShareObserver(observerFunc: (observer: ContentShareObserver) => void): void {
+    for (const observer of this.observerQueue) {
+      new AsyncScheduler().start(() => {
+        if (this.observerQueue.has(observer)) {
+          observerFunc(observer);
+        }
+      });
+    }
+  }
+
+  audioVideoDidStart(): void {
+    this.forEachContentShareObserver(observer => {
+      Maybe.of(observer.contentShareDidStart).map(f => f.bind(observer)());
+    });
+  }
+
+  audioVideoDidStop(sessionStatus: MeetingSessionStatus): void {
+    if (sessionStatus.statusCode() === MeetingSessionStatusCode.Left) {
```

Comment:
I would think we would want to still broadcast events even if the status code was something other than Left